src/components/withWindowDimensions.js

```diff
@@ -28,67 +28,33 @@ const windowDimensionsProviderPropTypes = {
     children: PropTypes.node.isRequired,
 };
 
-class WindowDimensionsProvider extends React.Component {
-    constructor(props) {
-        super(props);
+function WindowDimensionsProvider(props) {
+    const window = useWindowDimensions();
+    const insets = useSafeAreaInsets();
 
-        this.onDimensionChange = this.onDimensionChange.bind(this);
-
-        const initialDimensions = Dimensions.get('window');
-        const isSmallScreenWidth = initialDimensions.width <= variables.mobileResponsiveWidthBreakpoint;
-        const isMediumScreenWidth = initialDimensions.width > variables.mobileResponsiveWidthBreakpoint
-          && initialDimensions.width <= variables.tabletResponsiveWidthBreakpoint;
-        const isLargeScreenWidth = !isSmallScreenWidth && !isMediumScreenWidth;
-
-        this.dimensionsEventListener = null;
-
-        this.state = {
-            windowHeight: initialDimensions.height,
-            windowWidth: initialDimensions.width,
-            isSmallScreenWidth,
-            isMediumScreenWidth,
-            isLargeScreenWidth,
-        };
-    }
-
-    componentDidMount() {
-        this.dimensionsEventListener = Dimensions.addEventListener('change', this.onDimensionChange);
-    }
-
-    componentWillUnmount() {
-        if (!this.dimensionsEventListener) {
-            return;
-        }
-        this.dimensionsEventListener.remove();
-    }
-
-    /**
-     * Stores the application window's width and height in a component state variable.
-     * Called each time the application's window dimensions or screen dimensions change.
-     * @link https://reactnative.dev/docs/dimensions
-     * @param {Object} newDimensions Dimension object containing updated window and screen dimensions
-     */
-    onDimensionChange(newDimensions) {
-        const {window} = newDimensions;
+    const dimensions = useMemo(() => {
+        // On Android the window height does not include the status bar height, so we need to add it manually.
+        const windowHeight = Platform.OS === 'android'
```

Comment:
We don't generally allow `Platform.OS` checks. We create [platform specific files](https://github.com/Expensify/App#platform-specific-file-extensions). In this case we could create a util function instead of the whole component that either returns `window.height` or `window.height + insets.top` based on the platform type.